src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -39,6 +39,11 @@ class FrameworkExtension extends Extension
     private $sessionConfigEnabled = false;
 
     /**
+     * @var string|null
+     */
+    private $kernelRootHash;
```

Comment:
If I understand this right, the hash is just to have a unique ID for the kernel (in case there are several kernels in the app). Then, why not calling this variable `$kernelHash` or even `$kernelId`. This way the variable and method are "future proof" because we are not constrained to always use a hash and the location of the kernel.